### DISCUSSION TITLE
Clarification Needed on Upcoming API Changes Affecting Shipping Software Integration

### DISCUSSION THREAD

**Discussion by itiievskyi** [etsy/open-api#1303] [Q&A]
Hi everyone,

It has come to my attention that significant changes are happening with the Etsy API, particularly impacting shipping software like Pirate Ship. Starting from Monday, October 21, 2024, Etsy will reportedly remove the ability for third-party shipping software to access shipping addresses, submit tracking numbers, and automatically mark items as shipped. This has been communicated to users through partner shipping platforms and Etsy and in Etsy Seller Help Center (https://help.etsy.com/hc/en-us/articles/26654295371031-How-to-Use-a-Third-Party-Provider-to-Ship-Your-Order?segment=selling), but we couldn’t find any relevant documentation updates or announcements.

Could you please clarify the following:
1. Has there been a documentation update about these API changes? If so, can you point us to it?
2. Will existing connections to third-party services (like Pirate Ship) be impacted after the October 21 date?
3. What are the reasons behind these changes, and are there any alternative methods being suggested for handling order shipment data?

Any insights or updates on this would be greatly appreciated, as this has the potential to create manual work for many businesses relying on automated shipping processes.

Thank you in advance for your time and support.

**Comment by samborambo305** [comment]
This needs to be addressed. This seemingly arbitrary change impacts dozens of companies. @nateswart 

Our print-on-demand application requires these endpoints and is extremely useful for our Etsy users.

Also:
- We have our own Shipstation account and ALL of our shipment labels are purchased through them and on behalf of our POD clients.
- It makes no sense not to allow us to update this field for our customers via API. All this would do is introduce friction and annoy Etsy sellers.

**Reply by robneir** [reply]
+1 to this^

**Reply by sparkweb** [reply]
While it seems that this is targeted at third-party shipping label apps like PirateShip that provide a comparable service to Shippo/ShipStation, this is going to have the side-effect of completely destroying the print-on-demand ecosystem at Etsy. No new sellers will be able to use any third-party service to produce Print on Demand or fulfill any orders that might be stored at a warehouse. This seems very short-sighted on Etsy's part.

**Reply by amitkumar-248** [reply]
hello @itiievskyi 
Etsy has now [officially released the update](https://github.com/etsy/open-api/releases/tag/3.0.0-general-release-2024-10-21). Could you share your plans for tackling this change?

**Reply by itiievskyi** [reply]
From my perspective, there are a few options to handle this situation:

- Manual Copy-Paste: Clients can copy and paste addresses to create labels, similar to what PirateShip offers. While it's functional, this method can be frustrating and time-consuming for clients, especially with larger order volumes.
- Export Orders from Etsy: Clients can export their order data from Etsy and then import it into your system (or any shipping platform). This still requires some manual work but might be more manageable for clients used to copying and pasting. It offers bulk handling and may be a bit more efficient than handling orders one by one.
- Hope for Etsy to Restore Functionality: This is more of a long shot. Waiting for Etsy to reverse their decision and restore the missing functionality isn't something I'd bank on.
- Integration with Shippo: You could explore integrating with Shippo, but there are caveats. In this case, the client may need to connect their store twice to extract different sets of data. This might work if your system handles more than just orders, but it does add complexity and might feel redundant.

**Reply by turboturtleD** [reply]
> * Integration with Shippo: You could explore integrating with Shippo, but there are caveats. In this case, the client may need to connect their store twice to extract different sets of data. This might work if your system handles more than just orders, but it does add complexity and might feel redundant.

The same thing can be done with ShipStation.  This not a free solution, but if ShipStation is already being used, it’s a viable method.  You will also be able to access customer email addresses which had been removed from the Etsy API previously.

**Comment by chrisblach** [comment]
So much for an "Open API". They give only two companies a monopoly over shipping. This does not make software better. This restricts competition from improving the status quo.

**Reply by amitkumar-248** [reply]
Hi @nateswart, 
This restriction is hampering our business very badly, is it possible to get this restriction reverted or Exempted for all the existing apps.
 thanks

**Comment by displayground-josh** [comment]
Question: has anyone tested and confirmed that newly added integrations do fail when trying to pull shipping details or update an order?

Do we know what constitutes a "Etsy Preferred Partner or is in an approved category"?

**Comment by sen-gurkan** [comment]
Did you guys find any solution to this? Anyone knows what are the approved categories?

**Comment by swiftmyday** [comment]
Many people are leaving Etsy and closing their store for decisions like this. If they want to prevent drop shipping and non handmade stores, this is definitely not the correct way. Horrible decision.

**Comment by jamesholz** [comment]
Any updates for this? This is such an anti-consumer move.

**Comment by insiba** [comment]
The same limitation has been confirmed in our integration as well, and the impact is substantial for EU-based merchants.

For certain receipts (notably for sellers in Germany and France), all shipping-related fields are returned as `null`, including `country_iso`, even for physical orders (`is_digital = false`) with shipping costs and completed shipments. This indicates that the destination exists internally but is not exposed via the API.

From a technical and regulatory perspective, the absence of `country_iso` is not a minor inconvenience but a blocking issue. Within the European Union, the buyer’s country is a mandatory input for VAT determination. It defines the applicable VAT rate, the VAT regime (domestic, intra-EU, OSS, export), and is required for generating legally compliant invoices. Without this field, compliant invoicing is not possible.

The classification of the country as restricted personal data is difficult to reconcile. The country alone does not identify an individual, is not sensitive information, and is already implicitly processed within checkout, taxation, and shipping logic. Restricting this specific field does not meaningfully increase privacy protection, but it does prevent legally required business processes.

This has been raised with developer support multiple times. The responses consistently state that these fields are restricted for privacy reasons, without addressing the distinction between full address data and the non-identifying country field. Requests for escalation to a policy or compliance review have not led to a different outcome.

There are also broader ecosystem implications. The restriction effectively limits access to essential fulfillment data and makes it significantly harder for alternative providers to build compliant ERP and invoicing solutions. This reduces vendor choice for merchants and creates structural barriers for newer or more specialized tools, particularly those focused on EU VAT compliance and cross-border invoicing.

At minimum, access to `country_iso` would resolve the compliance issue without exposing sensitive address data.

**Comment by mistermannel** [comment]
Hi everyone,

Adding a different angle to this thread that I haven't seen raised yet: accounting and tax compliance, specifically for EU sellers.

The current restriction doesn't just affect shipping software and POD - it also blocks any third-party tool that needs billing addresses for legally required invoice generation.

The problem for German/EU sellers:
Under German tax law (§14 UStG) and EU VAT rules, every invoice must contain the customer's full billing address. This is not optional. Several scenarios make address data unavoidable:

- OSS (One-Stop-Shop): EU cross-border sales require destination-country VAT rates. The customer's country determines whether we charge 19% (DE), 20% (AT), 23% (PT), etc.
- Reverse Charge (B2B): Tax-exempt sales to EU businesses require full address documentation.
- Third-Country Exports: VAT-exempt sales outside the EU still require address proof for tax authorities.

Without address access, German Etsy sellers cannot produce compliant invoices through any third-party accounting tool. Their only option is manual copy-paste from the Etsy dashboard into accounting software like sevDesk, Lexware, or DATEV - for every single order.
This is a different category from shipping. We're not competing with Pirate Ship, Shippo, or ShipStation. We're not touching fulfillment. We need read-only access to billing addresses for invoice generation only - no createReceiptShipment, no tracking updates, no write access to orders.
I submitted a formal request (#23154031) three months ago outlining this use case. No response yet.

My ask to Etsy / @nateswart :
Could "tax and accounting compliance" be considered as a separate approved category, distinct from shipping fulfillment? The use case is narrow, legally mandated, and would meaningfully reduce administrative burden for tens of thousands of EU sellers. Happy to undergo any security review.

Would love to hear from other EU sellers on the thread - if you're affected by the same problem, please chime in so Etsy sees this isn't an edge case.

**Reply by mistermannel** [reply]
**Update (follow-up to my comment above):**

Quick update on the accounting/tax-compliance angle, because it's taken a turn that may interest others here.
After ~4 months, Etsy support told me in writing on 16 May that my app was "approved for the requested read-only billing address access." Great news - except the API never changed. Across 6 shops and 16 sampled receipts, all gated ShopReceipt address fields (first_line, city, zip, country_iso, formatted_address, etc.) still return null. Only name populates.
When I reported that, a later message said the request was actually "still under review" - directly contradicting the approval. Since then I've cycled through four different support agents, each promising an email follow-up. None has arrived. Status today: unchanged, and Etsy itself can't tell me whether I'm approved or not.
So two takeaways for anyone watching this thread:

Even when the use case is read-only, legally mandated (§14 UStG / EU VAT), and explicitly not shipping/fulfillment, there appears to be no working approval path - or at least no one who can confirm the provisioning step.
If you're an EU seller or build accounting tools for Etsy and are blocked by this, please comment. The more concrete this is, the harder it is to treat as an edge case.

@nateswart - is there an owner for developer address-access approvals who can speak to the provisioning step (backend flag vs. OAuth re-auth vs. additional scope)? A written approval that the system doesn't honor isn't an approval anyone can build on.

**Comment by Vocinglero** [comment]
French counterpart to @mistermannel's case, since he asked other EU developers to speak up. Same use case, different country.

I build an invoicing and VAT-compliance platform for online sellers in France. Read-only, invoice generation only: no `createReceiptShipment`, no tracking writes, no fulfillment. French invoicing rules require the customer's address on the invoice, and the buyer's country drives VAT determination (domestic, intra-EU, OSS, IOSS, export). Without these fields the integration cannot exist.

I opened this in May 2026 and have been through the support queue since, with contradictory answers along the way — told in August both that no path exists outside Preferred Partner status and, the next day, that the request was an active priority with the Partnership team. I won't belabour it; @mistermannel already documented the pattern.

I'd rather focus on the two questions that matter, and support @insiba's narrower proposal: if full address access is off the table, `country_iso` alone would unblock VAT determination for every EU seller without exposing anything that identifies an individual.

**@nateswart** — does an approved category for tax and accounting compliance exist today, or is Preferred Partner the only route? And who owns the provisioning step, given that a written approval did not change the API for @mistermannel? If there is no path for independent accounting tools, saying so in the docs would save the next developer months.

Other EU accounting developers blocked by this: please add your case.
